packages/react-components/src/components/GridLayout.tsx

```diff
@@ -31,43 +24,99 @@ export interface GridLayoutProps {
   styles?: BaseCustomStylesProps;
 }
 
-const calculateStandardLayoutRows = (numberOfItems: number, gridCol: number): number =>
-  Math.ceil(numberOfItems / gridCol);
-
-const calculateStandardLayoutColumns = (numberOfItems: number): number =>
-  numberOfItems > 0 ? Math.ceil(Math.sqrt(numberOfItems)) : 1;
-
 /**
- * A component to lay out audio / video tiles in a call.
+ * A component to lay out audio / video participants tiles in a call.
  *
  * @public
  */
 export const GridLayout = (props: GridLayoutProps): JSX.Element => {
-  const [gridCol, setGridCol] = useState(1);
-  const [gridRow, setGridRow] = useState(1);
-
-  const { children, layout = 'standard', styles } = props;
+  const { children, styles } = props;
   const numberOfChildren = React.Children.count(children);
 
-  switch (layout) {
-    case 'standard': {
-      const numberOfColumns = calculateStandardLayoutColumns(numberOfChildren);
-      if (gridCol !== numberOfColumns) setGridCol(numberOfColumns);
-      const numberOfRows = calculateStandardLayoutRows(numberOfChildren, gridCol);
-      if (gridRow !== numberOfRows) setGridRow(numberOfRows);
-      break;
+  const targetRef = useRef<HTMLDivElement>(null);
+  const [blockProps, setBlockProps] = useState<BlockProps>({
+    horizontal: true,
+    numBlocks: Math.ceil(Math.sqrt(numberOfChildren))
+  });
+
+  useEffect(() => {
+    const updateBlockProps = (): void => {
+      if (targetRef.current) {
+        setBlockProps(
+          calculateBlockProps(numberOfChildren, targetRef.current.offsetWidth, targetRef.current.offsetHeight)
+        );
+      }
+    };
+    const observer = new ResizeObserver(updateBlockProps);
+    if (targetRef.current) {
+      observer.observe(targetRef.current);
     }
+    updateBlockProps();
+    return () => observer.disconnect();
+  }, [numberOfChildren, targetRef.current?.offsetWidth, targetRef.current?.offsetHeight]);
+
+  return (
+    <div ref={targetRef} className={mergeStyles(gridLayoutContainerStyle, styles?.root)}>
+      {createBlocks({ ...blockProps, children })}
+    </div>
+  );
+};
+
+/**
+ * Props to create blocks for children in Grid Layout
+ *
+ */
+export type BlockProps = {
+  horizontal: boolean;
+  numBlocks: number;
+};
+
+const createBlocks = (props: BlockProps & { children: React.ReactNode }): React.ReactNode => {
+  const blocks: JSX.Element[] = [];
+  // Get the percent of 1 block out of the total number of blocks rounded down.
+  const blockPercent = (Math.floor(10000 / props.numBlocks) / 100).toFixed(2);
+  // Split array of children into as even chunks
+  const chunks = chunkify(React.Children.toArray(props.children), props.numBlocks);
+  for (let i = 0; i < props.numBlocks; i++) {
+    // Create block for i-th chunk children
+    const block = createBlock(props.horizontal, `block-${i}`, `${blockPercent}%`, chunks[i]);
+    // Add block to blocks
+    blocks.push(block);
   }
+  return blocks;
+};
 
+const createBlock = (
```

Comment:
It's a standard React component.
Let's name it as such

`Block` for example.

Reply:
Removed this because createBlocks was causing so much unnecessary re-rendering.